lib/variant.ml

```diff
@@ -72,7 +72,7 @@ let of_string s =
    | None -> s, `X86_64, Opam_version.default
    | Some (s, a) ->
       let arch a = Ocaml_version.of_opam_arch a |> Option.value ~default:`X86_64 in
-      match Astring.String.cut ~rev:true ~sep:"_opam-" a with
+      match Astring.String.cut ~rev:true ~sep:"opam-" a with
```

Comment:
The current fix only works for x86_64 variants.

I think the whole function would be a lot clearer if `String.split_on_char ’_’` was used instead.
Something like:
```
match String.split_on_char ’_’ s with
| [s] -> s, `X86_64, Opam_version.default
| [s; v] when String.is_prefix "opam-" v -> s, x86_64, Opam_version.of_string v
| [s; a] -> s, arch a, Opam_version.default
| [s; a; v] when String.is_prefix "opam-" v -> s, arch a, Opam_version.of_string v
```